extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/discovery/TradfriDiscoveryParticipant.java

```diff
@@ -37,6 +39,14 @@
 
     private static final String SERVICE_TYPE = "_coap._udp.local.";
 
+    /**
+     * RegEx patter to match the gateway name announced by mDNS
+     * Possible values:
+     * gw:001122334455, gw-001122334455, gw:00-11-22-33-44-55, gw-001122334455ServiceName
+     * 
+     */
+    private final String GATEWAY_NAME_REGEX_PATTERN = "gw[:-]{1}([a-f0-9]{2}[-]?){6}.*";
```

Comment:
Sorry, I do not understand how your regex corresponds to the examples given.
Can't the regex be much simpler then? Something like `gw[:-].*`?
How about adding some unit tests on this?

Reply:
Well - it can even be ```.*``` but I think it's wort at least checking that the mac address is present in the Service name. (Further more - the mac address check was already there - what I added was support for also a dash instead of colon after the ```gw``` and support for any string appearing after the mac address.

Reply:
The question is: Is there any definition by IKEA of how the name is actually constructed? Why allowing only arbitrary strings after the MAC but not in front of it? Do we have any other information in the service that can be used to clearly identify a Tradfri gateway?

Reply:
> Do we have any other information in the service that can be used to clearly identify a Tradfri gateway?

I don't have such - the change of the pattern was made after some debugging of the ServiceInfo message returned by the gateway. If you have any other information - please share it